app/models/post.rb

```diff
@@ -0,0 +1,13 @@
+class Post < ApplicationRecord
+  has_many :comments, foreign_key: 'post_id'
+  has_many :likes, foreign_key: 'post_id'
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+
+  def update_post_counter
+    author.increment!(:posts_counter)
+  end
+
```

Comment:
- [ ] **[OPTIONAL]** I think it would be better if we implemented an `after_save` call back for this method. So that the method is automatically called whenever a post is saved.

  ```
  after_save :update_post_counter

  ...  

  private
  
  def update_post_counter
    author.increment!(:posts_counter)
  end
  ```